test/unit/green-flag.test.jsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-env jest */
+const React = require('react'); // eslint-disable-line no-unused-vars
+const {shallow} = require('enzyme');
+const GreenFlag = require('../../src/containers/green-flag'); // eslint-disable-line no-unused-vars
+const renderer = require('react-test-renderer');
+const VM = require('scratch-vm');
+
+describe('GreenFlag', () => {
+    let vm;
+    beforeEach(() => {
+        // @todo: Ask scratch team their thoughts on mocking the VM.
+        vm = new VM();
+    });
+
+    test('renders active state', () => {
+        const component = renderer.create(
+            <GreenFlag active={true} vm={vm}/>
+        );
+        expect(component.toJSON()).toMatchSnapshot();
+    });
+
+    test('renders inactive state', () => {
+        const component = renderer.create(
+            <GreenFlag active={false} vm={vm}/>
+        );
+        expect(component.toJSON()).toMatchSnapshot();
+    });
+
+    test('triggers onClick when active', () => {
+        const onClick = jest.fn();
+        const componentShallowWrapper = shallow(
+            <GreenFlag active={true} onClick={onClick} vm={vm}/>
```

Comment:
So first thanks again for all of this, it is wonderful! 

One question that I have is how you feel / how the world feels about snapshot testing. Pros: it is insanely easy to implement. Cons: it hides what the "correct" state is, so you don't get the opportunity for your tests to feed backwards into the way you are writing the code (something that happens to me often when writing tests, i.e. if the test writes/reads badly then the code is usually not good). Do you have any opinion about snapshot testing?

As a next step, maybe let's try adding a test for another component / container pairing. I'm particularly interested in how the redux model of separating component / container plays out in testing, which is where it is meant to shine.